tests/unit/utils/checksum.spec.ts

```diff
@@ -0,0 +1,406 @@
+import { constants, createReadStream } from 'node:fs';
+import { Readable } from 'node:stream';
+import { Logger } from '@map-colonies/js-logger';
+import { trace, Tracer } from '@opentelemetry/api';
+import { Checksum } from '../../../src/utils/hash/checksum';
+import { ChecksumError } from '../../../src/ingestion/errors/ingestionErrors';
+import type { ChecksumProcessor } from '../../../src/utils/hash/interfaces';
+
+jest.mock('node:fs');
+jest.mock('@opentelemetry/api');
+
+describe('Checksum', () => {
+  let checksum: Checksum;
+  let mockLogger: jest.Mocked<Logger>;
+  let mockTracer: jest.Mocked<Tracer>;
+  let mockChecksumProcessor: jest.Mocked<ChecksumProcessor>;
+  let mockChecksumProcessorInit: jest.Mock;
+
+  beforeEach(() => {
+    mockLogger = {
```

Comment:
we dont have to to mock the logger method or test it just, mock it the normal way was we do every service:

`mockLogger` = jsLogger({enabled: false})

and remove all expectation of current logs methods